catalog/resource_lakehouse_monitor.go

```diff
@@ -0,0 +1,120 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const DefaultProvisionTimeout = 15 * time.Minute
+
+func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
+	return retry.RetryContext(ctx, DefaultProvisionTimeout, func() *retry.RetryError {
+		endpoint, err := w.LakehouseMonitors.GetByFullName(ctx, monitorName)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		if endpoint.Status == catalog.MonitorInfoStatusMonitorStatusActive {
+			return nil
+		}
+		if endpoint.Status == catalog.MonitorInfoStatusMonitorStatusError || endpoint.Status == catalog.MonitorInfoStatusMonitorStatusFailed {
+			return retry.NonRetryableError(fmt.Errorf("monitor status retrund %s for monitor: %s", endpoint.Status, monitorName))
+		}
+		return retry.RetryableError(fmt.Errorf("monitor %s is still pending", monitorName))
+	})
+}
+
+func ResourceLakehouseMonitor() common.Resource {
+	monitorSchema := common.StructToSchema(
+		catalog.MonitorInfo{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			common.CustomizeSchemaPath(m, "assets_dir").SetRequired()
+			common.CustomizeSchemaPath(m, "output_schema_name").SetRequired()
+			common.CustomizeSchemaPath(m, "table_name").SetRequired()
+			common.CustomizeSchemaPath(m).AddNewField("skip_builtin_dashboard", &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Required: false,
+			})
+			common.CustomizeSchemaPath(m).AddNewField("warehouse_id", &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Required: false,
+			})
+			common.CustomizeSchemaPath(m, "monitor_version").SetReadOnly()
+			common.CustomizeSchemaPath(m, "drift_metrics_table_name").SetReadOnly()
+			common.CustomizeSchemaPath(m, "profile_metrics_table_name").SetReadOnly()
+			common.CustomizeSchemaPath(m, "status").SetReadOnly()
+			common.CustomizeSchemaPath(m, "dashboard_id").SetReadOnly()
+			return m
+		},
+	)
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+
+			var create catalog.CreateMonitor
+			common.DataToStructPointer(d, monitorSchema, &create)
+			create.FullName = d.Get("table_name").(string)
```

Comment:
Isn't this name part of the Create response? How does the user know this ahead of time?

Reply:
Yes this is the `FullName` which is a path parameter. The name here is the same as the table name that the user wants to attach the monitor to. Therefore they do know this ahead of time.